systemd/systemd.go

```diff
@@ -117,6 +117,39 @@ func MockSystemctl(f func(args ...string) ([]byte, error)) func() {
 	}
 }
 
+type ServiceState struct {
+	ActiveState   string
+	UnitFileState string
+}
+
+// If `cmd` is the command issued by systemd.Status(), this function returns
+// the output to be produced by the command so that the queried services will
+// appear having the ActiveState and UnitFileState according to the data
+// passed in the `states` map.
+func HandleMockAllUnitsActiveOutput(cmd []string, states map[string]ServiceState) []byte {
```

Comment:
this is too specific and as doesn't really need anything private to this package, I think it should be moved to a new systemd/systemdtest package.

Reply:
mmm... OK :-)

I was actually [asked](https://github.com/snapcore/snapd/pull/10266#discussion_r633378014) to put this here, but I agree with your suggestion.